_posts/2021-01-08-2020-review.md

```diff
@@ -0,0 +1,254 @@
+  tools that need to inspect table metadata (e.g., DBeaver, Datagrip, etc).
+* [Faster queries on nested data in Parquet and ORC]({% post_url
+  2020-08-14-dereference-pushdown %})
+* Faster and more accurate `approx_percentile`, based on t-digest data structure
+* Support of Bloom filters in ORC
+* Optimized Parquet writer
+
+## Security
+
+The more data you access with Trino, the more it becomes critical to secure it.
+With that in mind we added a lot of improvements:
+
+* The [Web UI](/docs/current/admin/web-interface.html) now requires
+  authentication. Various actions such as viewing query details, killing
+  queries, etc., are protected with authorization checks based on the identity
+  of the user. Additionally, the UI now supports OAuth2 for user identification.
+* External and internal APIs are now properly secured with authentication and
+  authorization checks. Importantly, this fixes a [CVE reported
+  vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15087)
+  that affects in all older and other versions of Presto®]
+*  A [new mechanism to externalize secrets in configuration
+   files](/docs/current/security/secrets.html) that makes it easier to integrate
+   with third-party secret managers and deployment tools.
+* Support for JSON Web Key (JWK) authentication and [pluggable certificate
+  authenticators](/docs/current/develop/certificate-authenticator.html)
+* [Salesforce authenticator](docs/current/security/salesforce.html)
+* The query engine and access control SPIs now support injecting row filters and
+  column masks.
+* New syntax for managing permissions (`GRANT/REVOKE` on schema,
+  `ALTER TABLE/SCHEMA/VIEW ... SET AUTHORIZATION`)
+
+## Supported data source
+
+Trino empowers you to use one platform to access all data sources. Connectors
+enable this and we added numerous new connectors:
+
+* [Iceberg](/docs/current/connector/iceberg.html)
+* [Prometheus](/docs/current/connector/prometheus.html)
+* [Oracle](/docs/current/connector/oracle.html)
+* [Pinot](/docs/current/connector/pinot.html)
+* [Druid](/docs/current/connector/druid.html)
+* [BigQuery](/docs/current/connector/bigquery.html)
+* [MemSQL](/docs/current/connector/memsql.html)
+
+All other connectors received a large host of improvements. Let’s just look at
+two popular connectors:
+
+### Hive connector for HDFS, S3 and cloud object storage system
+
+* Complex Hive views, allows seamless integration with Hive or simplifying
+  migration from Hive
+* [ACID transactional tables]({% post_url 2020-06-01-hive-acid %}) with `INSERT`
+  and `DELETE` support
+* [Built-in storage caching](/docs/current/connector/hive-caching.html) and
+  support for [external caching with
+  Alluxio](/docs/current/connector/hive-alluxio.html)
+* New procedures: `system.drop_stats()`, `register_partition()`,
+  `unregister_partition()`
+* Support for [Azure object storage](/docs/current/connector/hive-azure.html)
```

Comment:
i am not sure when it was added. was it not 2019?

Reply:
Martin found it in 2019